client/solve.go

```diff
@@ -423,13 +423,13 @@ func parseCacheOptions(opt SolveOpt) (*cacheOptions, error) {
 					continue
 				}
 				for _, m := range idx.Manifests {
-					if m.Annotations[ocispec.AnnotationRefName] == "latest" {
+					if m.Annotations[ocispec.AnnotationRefName] == "latest" || (attrs["tag"] != "" && m.Annotations[ocispec.AnnotationRefName] == attrs["tag"]) {
```

Comment:
Why do we still match "latest" when a custom tag is set by the user?

Reply:
@tonistiigi 
Thank you for the review.
The "latest" comparison was for cases where the custom tag was not set (to preserve the same behavior as before).
However, if a custom tag has been set, you need to change the order of comparison because you do not need to compare it with "latest".

Reply:
I might be missing something but shouldn't it be  `if (m.Annotations[ocispec.AnnotationRefName] == "latest" && attrs["tag"] == "") || (attrs["tag"] != "" && m.Annotations[ocispec.AnnotationRefName] == attrs["tag"]) {`

Reply:
@tonistiigi 

> I might be missing something but shouldn't it be if (m.Annotations[ocispec.AnnotationRefName] == "latest" && attrs["tag"] == "") || (attrs["tag"] != "" && m.Annotations[ocispec.AnnotationRefName] == attrs["tag"]) {

Could you give me a little more detail as I don't yet understand why? (I pushed the changes, but please ignore them.)

Reply:
If the user sets a tag then `latest` should not be matched any more. At least this is how I understand it.

Reply:
> If the user sets a tag then latest should not be matched any more. At least this is how I understand it.

Certainly, if a tag is set, it will not match the latest one.I suppose that if you have a custom tag, you can use it and if not, make a change that only checks for latest.
`(attrs["tag"] != "" && m.Annotations[ocispec.AnnotationRefName] == attrs["tag"]) ||  m.Annotations[ocispec.AnnotationRefName] == "latest"`

Reply:
> Certainly, if a tag is set, it will not match the latest one.I

Sorry for wasting time on this but I fail to see how the current code matches that logic. If the manifest has latest annotation it will be matched independently from if a custom tag was set or not.

Reply:
> If the manifest contains the latest annotations, they will match regardless of whether custom tags are set.

With that in mind, we believe that behavior needs to be changed slightly.
If the manifest has the latest annotations and a custom tag is specified, we are considering changing to output an error message. How about that.

Reply:
> If the manifest has the latest annotations and a custom tag is specified, we are considering changing to output an error message. How about that.

Why? Isn't it a valid case to have latest that and other tags and choose between them.

@AkihiroSuda Am I missing something here?

Reply:
@tonistiigi 

> If the manifest has the latest annotations and a custom tag is specified, we are considering changing to output an error message. How about that.

The information I gave was incorrect.  I thought that it would be better to issue an error message when the latest annotation is present in the manifest and an argument specifying a custom tag is specified.